csrc/device_lower/pass/allocation.cpp

```diff
@@ -592,15 +588,80 @@ class AllocationInserter : public kir::ExprMutator {
       }
     }
 
-    // Allocate mbarrier for cp.async.bulk, note that this is only a temporary
-    // solution, we should remove this after we have a better way to handle
-    // synchronizations for cp.async.bulk.
-    if (ir_utils::isCpAsyncBulkLoad(expr)) {
-      if (circular_buffer_depth > 1) {
+    // Allocate mbarrier for cp.async.bulk, for non-circular buffered cases by
+    // lowering a single cp.async.bulk as:
+    //    alloc mbarrier
+    //    init mbarrier
+    //    block_sync
+    //    cp.async.bulk
+    //    inval mbarrier
+    //    block_sync
+    // Note that this is only a temporary solution, we should remove this after
```

Comment:
I don't really know what is a good strategy to assign ldst with mbarrier. It depends on how we will use TMA. Let me remove this since there is no plan to improve it before we have a use case.

Reply:
Will do in https://github.com/NVIDIA/Fuser/pull/3219